vmdb/app/models/miq_provision_microsoft_workflow.rb

```diff
@@ -0,0 +1,63 @@
+class MiqProvisionMicrosoftWorkflow < MiqProvisionInfraWorkflow
+  def self.default_dialog_file
+    'miq_provision_dialogs'
+  end
+
+  def self.default_pre_dialog_file
+    'miq_provision_dialogs_pre'
+  end
+
+  def self.encrypted_options_fields
+    super + [:sysprep_password, :sysprep_domain_password]
+  end
+  def dialog_name_from_automate(message = 'get_dialog_name')
+    super(message, {'platform' => 'microsoft'})
+  end
+
+  def allowed_provision_types(_options = {})
+    {
+      "microsoft" => "Microsoft"
+    }
+  end
+
+  def self.allowed_templates_vendor
+    'microsoft'
+  end
+
+  def update_field_visibility(_options = {})
+    super
+
+    if get_value(@values[:vm_dynamic_memory])
+      show_fields(:edit, [:vm_minimum_memory, :vm_maximum_memory])
+    end
```

Comment:
I think you are missing an else case to hide the fields.  Otherwise once they are visible unchecking the dynamic_memory field will not hide the values unless the view code is already handling that for you.